plugins/gtav/Game.cpp

```diff
@@ -5,113 +5,79 @@
 
 #include "Game.h"
 
-#include "mumble_positional_audio_utils.h"
+#include <sstream>
 
-#include <cstring>
+Game::Game(const uint64_t id, const std::string &name) : m_proc(static_cast< procid_t >(id), name) {
+}
```

Comment:
Why do we pass the ID as `uint64_t` but store it as a `procid_t`? Why not consistently use the latter?

Reply:
The plugin API uses `uint64_t`, but `procid_t` is currently defined as `uint32_t`.

We have to define `procid_t` as `uint64_t`.

Reply:
That'd work. But if we want to cast, I think we should cast at the point where we are calling the constructor. Imo the arguments of a constructor should match the data that it really expects. If it uses different data types then that usually signifies something particular (e.g. that instead of a simple cast some extra operation has to be performed) :thinking:

Reply:
We would have to edit both `gtav.cpp` and `Game.cpp` when we change the `procid_t` definition though.

Reply:
Yes, at this point it would only be for consistency with the Among Us plugin:

https://github.com/mumble-voip/mumble/blob/dd5df45313a937359285327f556557159bab758e/plugins/amongus/Game.cpp#L10

Reply:
Well we could just change the Among Us plugin as well, can't we?

Reply:
What's the point if we're going to change the definition of `procid_t`?

Reply:
We don't need the cast then, so we should remove it

Reply:
Yes - my point is that we should just change it within this PR. Otherwise it will only be forgotten.
So simply add a separate REFAC commit that does exactly this and everything should be fine.

Reply:
Ah, okay, good idea.